docs/registry/create/_index.md

```diff
@@ -11,154 +11,109 @@ tags:
+A module provides one or more {{< glossary_tooltip term_id="modular-resource" text="modular resources" >}}, and is packaged in a manner that streamlines deployment to a Viam machine.
+Modules run alongside [`viam-server`](/get-started/installation/) as separate processes, communicating with `viam-server` over UNIX sockets.
+When a module initializes, it registers its {{< glossary_tooltip term_id="model" text="model or models" >}} and associated [APIs](/build/program/apis/) with `viam-server`, making the new model available for use.
 
-Modules run alongside `viam-server` as separate processes, communicating with `viam-server` over UNIX sockets.
-When a module initializes, it registers its {{< glossary_tooltip term_id="model" text="model or models" >}} and the associated [APIs](/build/program/apis/) with `viam-server`, making the new model available for use.
+You can search the [Viam Registry](https://app.viam.com/registry) and [deploy an existing module to your machine](/registry/configure/) in a few clicks if you find one that meets your needs.
+Or you can write your own module to address your specific use case, and either upload it to the Viam registry to share with others, or deploy it to your machine as a local module without uploading to the registry.
 
-In most cases, the {{< glossary_tooltip term_id="modular-resource" text="modular resources" >}} you create should be a new {{< glossary_tooltip term_id="model" text="model" >}} that implements an existing Viam [API](/build/program/apis/).
-For example, if you have a custom base, you can create a new base model that implements the `rdk:component:base` API.
+Follow the instructions below to learn how to write a new module using your preferred language and its corresponding [Viam SDK](/sdks/).
 
-## Find your reference files
+## Overview of a module
 
-Before you start coding a new resource model, follow these steps to find the appropriate source code as a reference for the methods you need to implement:
+Generally, to write a module, you:
 
-1. **Understand your model's client interface:**
+1. [Define a new resource model](#define-a-new-resource-model) to define all the capabilities of your model.
+1. [Write an entry point (main program) file](#write-an-entry-point-main-program-file) to serve as the central interface to those new capabilities.
+1. [Compile or package](#compile-or-package-your-module) the model definition file or files, main program file, and any supporting files into a single executable file (a module) that can be run by `viam-server`.
 
-   {{< tabs >}}
-   {{% tab name="Python" %}}
+While you can certainly combine the resource model definition and the main program code into a single file if desired (for example, a single `main.py` program that includes both the model definition and the `main()` program that uses it), this guide will use separate files for each.
```

Comment:
> this guide will use separate files for each

Do we have a reason for this that we can share? Is it for clarity or separation of concerns?

Reply:
_Theory:_ The latter, great insight. My thinking:
- Keeping a clear path forward from concept to concept, as much as we can here.
- While some of our most simple examples across our repos do use one file for both 1 and 2, all of the more complicated ones use two. Starting with this file structure from the beginning might set users up for success when they later transition to module design that accomplished more complicated tasks, like a component and a service working together.
- This also helps us to demonstrate including other files and referencing the methods/functions contained within them.

_Practice:_ But, mostly, at the end of the day: our existing sample code for all three languages was split in this fashion. Refactoring to a single file for each would be much work, and for minimal gain.

But I will add context to this statement here for the end user. Thanks!